app/scripts/move-popup/dimMoveItemProperties.directive.js

```diff
@@ -165,12 +162,17 @@
       if (vm.item.primStat.statHash === 3897883278) {
         // it's armor.
         vm.light += ' Defense';
-        vm.classType = vm.item.classTypeName[0].toUpperCase() + vm.item.classTypeName.slice(1);
       } else if (vm.item.primStat.statHash === 368428387) {
         // it's a weapon.
         vm.light += ' Attack';
         vm.classes['is-' + vm.item.dmg] = true;
       }
+      if (vm.item.classTypeName !== 'unknown' &&
+          // These already include the class name
+          vm.item.type !== 'ClassItem' &&
+          vm.item.type !== 'Artifact') {
+        vm.classType = vm.item.classTypeName[0].toUpperCase() + vm.item.classTypeName.slice(1);
```

Comment:
I think modifying buolt-ins means you can't use optimized versions of those
objects.

On Sunday, July 31, 2016, kyle shay <notifications@github.com> wrote:

> In app/scripts/move-popup/dimMoveItemProperties.directive.js
> <https://github.com/DestinyItemManager/DIM/pull/730#discussion_r72906996>:
>
> >        } else if (vm.item.primStat.statHash === 368428387) {
> >          // it's a weapon.
> >          vm.light += ' Attack';
> >          vm.classes['is-' + vm.item.dmg] = true;
> >        }
> > +      if (vm.item.classTypeName !== 'unknown' &&
> > +          // These already include the class name
> > +          vm.item.type !== 'ClassItem' &&
> > +          vm.item.type !== 'Artifact') {
> > +        vm.classType = vm.item.classTypeName[0].toUpperCase() + vm.item.classTypeName.slice(1);
>
> why doesn't JS/es6 have a .toCapitalCase or something :( shame extending
> the string prototype is not an ideal solution... and is actually quite slow
> to boot. Maybe due to supporting multi locale? :/
>
>
> <https://camo.githubusercontent.com/551d35255da63d6166a6572a33e01664ba2ac18b/687474703a2f2f692e737461636b2e696d6775722e636f6d2f744e774b6b2e706e67>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/DestinyItemManager/DIM/pull/730/files/add63f1f03287309843bdb6082dc2f99580c437b#r72906996>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AATHeBlrI2KkUYwcq69MJ4N7n7cuL2-fks5qbNB-gaJpZM4JY83I>
> .
>